blark/transform.py

```diff
@@ -1621,6 +1622,12 @@ def from_lark(
         first_parameter: Optional[ParameterAssignment] = None,
         *remaining_parameters: ParameterAssignment,
```

Comment:
_Technically_ the `remaining_parameters` annotation needs updating here because it can now contain a `lark.Token` as well (holding that semi-colon `";"`). Then static analyzers will complain that `parameters` is no longer compatible with `List[ParameterAssignment]`, ...

Let's just roll with this and maybe clean it up later - there are much bigger fish to fry!

Reply:
I think that's a fair point, though, @klauer... I had toyed with it, but decided the same about "bigger fish" (at least for this moment in time). Do you have thoughts on a better name? I'd be happy to lob something in there in upcoming changes.

Reply:
I couldn't think of a way to express it in words - sometimes code is more succinct.
I think this might be preferable:
```python
    @staticmethod
    def from_lark(
        name: SymbolicVariable,
        *params: Union[ParameterAssignment, lark.Token, None],
    ) -> FunctionCall:
        # Condition parameters (which may be `None`) to represent empty tuple
        if params and params[0] is None:
            params = params[1:]
        dereferenced = bool(params and params[-1] == "^")
        if dereferenced:
            params = params[:-1]

        return FunctionCall(
            name=name,
            parameters=typing.cast(List[ParameterAssignment], list(params)),
            dereferenced=dereferenced,
        )

```

Reply:
OK! I'm gonna roll this in to the next bit of work I do. :+1: